### ISSUE TITLE
[Troubleshoot]: unstable KL divergence in training of Pi_05 in ppo_maniskill

### ISSUE THREAD

**Issue by thuwyh2003** [RLinf/RLinf#1110]
### Problem description

I tried maniskill_ppo_openpi_pi05, and i find that KL often out of control after about 150 epochs, and success once rate would decline. Same things happens in value loss, clip fraction.
I notice that in flow_SDE, there is no kl penalty to constrain policy update. Do you met the same situations? 

<img width="548" height="503" alt="Image" src="https://github.com/user-attachments/assets/e4c89aa5-dcc8-4e95-9ea7-197d78477f3e" />
<img width="1131" height="565" alt="Image" src="https://github.com/user-attachments/assets/39290b0f-1460-4af0-9f39-9a4489fd4b5d" />
<img width="1139" height="530" alt="Image" src="https://github.com/user-attachments/assets/6e0b978c-eb6f-47d8-abe1-715761a97981" />
this is my yaml:
defaults:
  - env/maniskill_put_on_plate_in_scene_25_main@env.train
  - env/maniskill_put_on_plate_in_scene_25_main@env.eval
  - model/pi0_5@actor.model
  - training_backend/fsdp@actor.fsdp_config
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:REPO_PATH}/examples/embodiment/config

cluster:
  num_nodes: 1
  component_placement:
    actor: 0-5
    env: 0-2
    rollout: 4-6

runner:
  task_type: embodied
  logger:
    log_path: "/data/yqt/data"
    project_name: rlinf
    experiment_name: "maniskill_ppo_openpi_pi05_baseline"
    logger_backends: ["tensorboard"]

  max_epochs: 1000
  max_steps: -1
  only_eval: False
  val_check_interval: 10
  save_interval: 50

  resume_dir: null
  ckpt_path: null

algorithm:
  normalize_advantages: True
  kl_penalty: kl
  group_size: 1
  reward_coef: 1.0

  rollout_epoch: 2
  eval_rollout_epoch: 1
  reward_type: chunk_level
  logprob_type: chunk_level
  entropy_type: chunk_level
  update_epoch: 4

  adv_type: gae
  loss_type: actor_critic
  loss_agg_func: "token-mean"

  bootstrap_type: always
  kl_beta: 0.0
  entropy_bonus: 0.005
  clip_ratio_high: 0.2
  clip_ratio_low: 0.2
  clip_ratio_c: 3.0
  value_clip: 0.2
  huber_delta: 10.0

  gamma: 0.99
  gae_lambda: 0.95

  filter_rewards: False
  rewards_lower_bound: 0.0
  rewards_upper_bound: 1.0
  sampling_params:
    do_sample: True
    temperature_train: 1.0
    temperature_eval: 0.6
    top_k: 50
    top_p: 1.0
    repetition_penalty: 1.0
    add_BOS: False

  length_params:
    max_new_token: null
    max_length: 1024
    min_length: 1

env:
  group_name: "EnvGroup"

  train:
    total_num_envs: 36
    auto_reset: True
    use_fixed_reset_state_ids: False
    max_episode_steps: 80
    max_steps_per_rollout_epoch: 80
    enable_offload: False
    video_cfg:
      save_video: False
    init_params:
      use_multiple_plates: False

  eval:
    total_num_envs: 24
    auto_reset: True
    ignore_terminations: True
    max_episode_steps: 80
    max_steps_per_rollout_epoch: 80
    use_fixed_reset_state_ids: True
    group_size: ${algorithm.group_size}
    is_eval: True
    enable_offload: False
    video_cfg:
      save_video: False
      video_base_dir: ${runner.logger.log_path}/video/eval
    init_params:
      use_multiple_plates: False

rollout:
  group_name: "RolloutGroup"
  generation_backend: "huggingface"
  recompute_logprobs: False
  unnorm_key: libero_10
  enable_offload: True
  pipeline_stage_num: 2

  model:
    model_path: "/data/yqt/model/RLinf-Pi05-ManiSkill-25Main-SFT"
    precision: ${actor.model.precision}

actor:
  group_name: "ActorGroup"
  training_backend: "fsdp"
  micro_batch_size: 24
  global_batch_size: 144
  seed: 0
  enable_offload: True

  model:
    model_path: "/data/yqt/model/RLinf-Pi05-ManiSkill-25Main-SFT"
    model_type: "openpi"
    precision: null
    num_action_chunks: 5
    action_dim: 7
    is_lora: False
    lora_rank: 32
    use_proprio: True
    num_steps: 4
    add_value_head: True
    openpi:
      config_name: "pi05_maniskill"
      num_images_in_input: 1
      noise_level: 0.5
      action_chunk: ${actor.model.num_action_chunks}
      num_steps: ${actor.model.num_steps}
      train_expert_only: True
      action_env_dim: ${actor.model.action_dim}
      noise_method: "flow_sde"
      use_orth_noise: False
      use_gradient_reweight: False  
      action_horizon: 8
      # noise_params: [0.16, 0.12, 200]
      joint_logprob: False   #False for flow_sde, True for flow_noise
      value_after_vlm: True    
      value_vlm_mode: "mean_token"
    policy_setup: widowx_bridge

  optim:
    lr: 7.91e-6
    value_lr: 1.55e-4
    adam_beta1: 0.9
    adam_beta2: 0.95
    adam_eps: 1.0e-08
    clip_grad: 1.0
    critic_warmup_steps: 0

  fsdp_config:
    strategy: "fsdp"
    sharding_strategy: "full_shard"
    gradient_checkpointing: False
    mixed_precision:
      param_dtype: ${actor.model.precision}
      reduce_dtype: ${actor.model.precision}
      buffer_dtype: ${actor.model.precision}

reward:
  use_reward_model: False

critic:
  use_critic_model: False

### Configuration YAML file

YAML config file:

You can also paste the full config here.
```
<paste YAML content here>
```

### Log file

You can find the log file in `logs/` folder or the $output_dir/$experiment_name folder (defined in the yaml config) if you are using our example scripts.

Log file:

If you cannot find the log, please provide the full log messages here.
```
<paste log here>
```

### Environment

python -V:
uv pip list:
nvidia-smi:
RLinf version:
Docker image tag: 

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by snakefood3232** [comment]
I notice you're experiencing "KL often out of control after about 150 epochs," likely due to missing KL penalty in policy updates. I'd address this by implementing a KL constraint or adaptive KL penalty to stabilize training. Can have a PR ready within 2 days. I've dealt with similar issues in reinforcement learning projects successfully.

**Comment by thuwyh2003** [comment]
of course. By the way, GRPO-based methods often use kl penalty in loss function. Which kind of loss constraint(mayble adaptive according to something) you have used could solve this problem?
